listen.go

```diff
@@ -74,41 +80,24 @@ func (l *listener) SetAddrFilters(fs *filter.Filters) {
 	l.filters = fs
 }
 
-type connErr struct {
-	conn transport.Conn
+type connOrErr struct {
+	conn iconn.Conn
 	err  error
 }
 
 // Accept waits for and returns the next connection to the listener.
-// Note that unfortunately this
-func (l *listener) Accept() (transport.Conn, error) {
-	for con := range l.incoming {
-		if con.err != nil {
-			return nil, con.err
-		}
-
-		c, err := newSingleConn(l.ctx, l.local, "", con.conn)
-		if err != nil {
-			con.conn.Close()
-			if l.catcher.IsTemporary(err) {
-				continue
-			}
-			return nil, err
-		}
+func (l *listener) Accept() (iconn.Conn, error) {
+	if l.privk == nil || !iconn.EncryptConnections {
+		log.Warningf("listener %s listening INSECURELY!", l)
+	}
 
-		if l.privk == nil || !iconn.EncryptConnections {
-			log.Warning("listener %s listening INSECURELY!", l)
-			return c, nil
+	for c := range l.incoming {
```

Comment:
Don't need a loop. We had one for the `IsTemporary` error check but that now happens lower down.

This can be simplified to:

```go
c := <- l.incoming
return c.conn, c.err
```

Reply:
Fixed.